modules/core/src/main/scala/dev/profunktor/redis4cats/caching/ClientSideCaching.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018-2021 ProfunKtor
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.profunktor.redis4cats.caching
+
+import cats.effect.kernel.{ Async, Resource }
+import dev.profunktor.redis4cats.effect.TxExecutor
+import io.lettuce.core.TrackingArgs
+import io.lettuce.core.api.StatefulRedisConnection
+import io.lettuce.core.support.caching.{
+  CacheAccessor => JCacheAccessor,
+  CacheFrontend => JCacheFrontend,
+  ClientSideCaching => JClientSideCaching
+}
+
+object ClientSideCaching {
+
+  def make[F[_]: Async, K, V](
+      connection: StatefulRedisConnection[K, V],
+      args: TrackingArgs,
```

Comment:
the convention in this lib , is to create a scala api for the java pojo's . [i.e.](https://github.com/profunktor/redis4cats/blob/dd861e5f2cef45b2b0cfcf1aab0a5af5968f48a1/modules/effects/src/main/scala/dev/profunktor/redis4cats/effects.scala#L216)